rotkehlchen/db/dbhandler.py

```diff
@@ -706,18 +707,27 @@ def get_dynamic_cache(
     ) -> Timestamp | None:
         ...
 
+    @overload
+    def get_dynamic_cache(
+            self,
+            cursor: 'DBCursor',
+            name: Literal[DBCacheDynamic.EXTRA_INTERNAL_TX],
+            **kwargs: Unpack[ExtraTxArgType],
+    ) -> ChecksumEvmAddress | None:
+        ...
+
     def get_dynamic_cache(
             self,
             cursor: 'DBCursor',
             name: DBCacheDynamic,
             **kwargs: str,
-    ) -> Timestamp | None:
+    ) -> Timestamp | ChecksumEvmAddress | None:
         """Returns the cache value from the `key_value_cache` table of the DB
         according to the given `name` and `kwargs`. Defaults to `None` if not found."""
         value = cursor.execute(
             'SELECT value FROM key_value_cache WHERE name=?;', (name.get_db_key(**kwargs),),
         ).fetchone()
-        return None if value is None else Timestamp(int(value[0]))
+        return None if value is None else value[0]
```

Comment:
this changes logic here, not only typing. If returning a timestamp from string this used to work and now it will no longer work. That is dangerous. What are the possibilities here?

Reply:
It's hard to track the possibilities. Maybe we can handle the typing similar to DBSettings like this?
```py
    def get_typed_value(self, value: str) -> Timestamp | ChecksumEvmAddress:
        """Returns the given value with the correct type for the current key"""
        if self == DBCacheDynamic.EXTRA_INTERNAL_TX:
            return string_to_evm_address(value)
        else:
            return Timestamp(int(value))
...
return None if value is None else name.get_typed_value(value[0])
```